basic-services/ReadPropertyResult.cs

```diff
@@ -39,42 +40,130 @@ public struct ReadPropertyResult
         // Helper 
         public bool IsReliable => Reliability == Reliable;
 
-        internal ReadPropertyResult(JToken token, CultureInfo cultureInfo, string reliability = null, string priority = null)
+        private CultureInfo _CultureInfo;
+
+        internal ReadPropertyResult(Guid id, JToken token, string attribute, CultureInfo cultureInfo = null, string reliability = null, string priority = null)
         {
+            Id = id;
+            Attribute = attribute;
             Reliability = reliability ?? Reliable;
             Priority = priority;
-            
+            StringValue = null;
+            NumericValue = 1;
+            ArrayValue = null;
+            BooleanValue = false;
+            _CultureInfo = cultureInfo;
+
+            ReadToken(token);
+        }
+
+        /// <summary>Parses the JToken type and assigns struct values appropriately.</summary>
+        internal void ReadToken(JToken token)
```

Comment:
Nitpick: I'd expect a ReadToken to return whatever was read. In this case you are really "processing the token", so I'd change all of the names of these methods to `ProcessXXXX` instead.

Reply:
Another thing to do is mark all of the process methods as private. Why? 

1. `internal` means that any other class in this dll can call that method. Which means that after you create an instance some of your code could call one of these methods and change the property values of an instance.
1. You want structs/classes like this to be immutable, meaning their value can't change after they are instantiated.

So it's okay from an organization perspective to separate out this methods, but mark them private so there's no way for them to ever be called after creation.